src/main/scala/mesosphere/marathon/core/volume/providers/DVDIProvider.scala

```diff
@@ -0,0 +1,245 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord.{ Validator, _ }
+import com.wix.accord.dsl._
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Environment, Volume => MesosVolume }
+
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected[volume] case object DVDIProvider
+    extends AbstractPersistentVolumeProvider("dvdi")
+    with OptionSupport {
```

Comment:
I don't like to mix in implementation with traits. (there is a reason why languages like golang/rust don't allow that anymore)

You can make OptionSupport simply an object and use an import. I even prefer referring to the types it provides with a prefix, e.g. `OptionSupport.NamedOption` which makes it easy to determine where it is coming from.